crates/bevy_render/src/camera/camera.rs

```diff
@@ -372,18 +370,23 @@ pub fn camera_system<T: CameraProjection + Component>(
         })
         .collect();
 
-    let mut added_cameras = vec![];
-    for entity in &queries.p1() {
-        added_cameras.push(entity);
-    }
-    for (entity, mut camera, mut camera_projection) in &mut queries.p0() {
+    for (mut camera, mut camera_projection) in &mut cameras {
         if camera
             .target
             .is_changed(&changed_window_ids, &changed_image_handles)
-            || added_cameras.contains(&entity)
+            || camera.is_added()
             || camera_projection.is_changed()
+            || camera.computed.old_viewport
+                != camera
+                    .viewport
+                    .as_ref()
+                    .map(|viewport| (viewport.physical_position, viewport.physical_size))
         {
             camera.computed.target_info = camera.target.get_render_target_info(&windows, &images);
+            camera.computed.old_viewport = camera
+                .viewport
+                .as_ref()
+                .map(|viewport| (viewport.physical_position, viewport.physical_size));
```

Comment:
I guess this value could be calculated once and used for both the logic and this assignment?

Reply:
Ping @jakobhellermann